server/src/services/tooljet_db.service.ts

```diff
@@ -249,25 +261,27 @@ export class TooljetDbService {
       referenced_tables_info = await this.fetchAndCheckIfValidForeignKeyTables(
         referenced_table_list,
         organizationId,
-        'TABLENAME'
+        'TABLENAME',
+        appManager
       );
     }
 
     const isFKfromCompositePK = await this.checkIfForeignKeyReferencedColumnsAreFromCompositePrimaryKey(
       foreign_keys,
-      organizationId
+      organizationId,
+      connectionManagers
     );
 
     if (isFKfromCompositePK)
       throw new ConflictException(
         'Foreign key cannot be created as the referenced column is in the composite primary key.'
       );
 
-    const queryRunner = this.manager.connection.createQueryRunner();
+    const queryRunner = appManager?.queryRunner || appManager.connection.createQueryRunner();
```

Comment:
Can't we always call `.connection.createQueryRunner();` ? 
Such that we can then have that specific query released without having to check `transactionDepth`? 🤔